pkg/ast/spl/spl.peg

```diff
@@ -5203,8 +5221,148 @@ MvexpandLimit <- SPACE "limit" EQUAL intValue:IntegerAsString {
+    numOptions := 1 + len(restSlice)
+
+    inputLookupOption := &structs.InputLookup{}
+
+    for i := 0; i < numOptions; i++ {
+        var inputLookupOptionArg *InputLookupOptionArgs
+        if i == 0 {
+            inputLookupOptionArg = first.(*InputLookupOptionArgs)
+        } else {
+            separator := restSlice[i - 1].([]any)
+            inputLookupOptionArg = separator[1].(*InputLookupOptionArgs)
+        }
+
+        argOption := inputLookupOptionArg.argOption
+        _, exist := optionWasSpecified[argOption]
+        if exist {
+            return nil, fmt.Errorf("Invalid Syntax, option: %v is already specified", argOption)
+        }
+        optionWasSpecified[argOption] = struct{}{}
+
+        switch argOption {
+            case "append":
+                inputLookupOption.Append = inputLookupOptionArg.inputLookupOption.Append
+            case "strict":
+                inputLookupOption.Strict = inputLookupOptionArg.inputLookupOption.Strict
+            case "start":
+                inputLookupOption.Start = inputLookupOptionArg.inputLookupOption.Start
+            case "max":
+                inputLookupOption.Max = inputLookupOptionArg.inputLookupOption.Max
+            default:
+                return nil, fmt.Errorf("Not a Valid InputLookup Option: %v", argOption)
+        }
+    }
+
+    return inputLookupOption, nil
+}
+
+
+
+InputLookupBlock <- PIPE CMD_INPUTLOOKUP inputLookupOption:(InputLookupOptionList) SPACE filename:(String) whereClause:(OptionalWhereClause)? {
+    inputLookup := inputLookupOption.(*structs.InputLookup)
+    if whereClause != nil {
+        inputLookup.WhereExpr = whereClause.(*structs.BoolExpr)
+    }
+    inputLookup.Filename = removeQuotes(filename.(string))
+    if inputLookup.Max == 0 {
+        inputLookup.Max = 1000000000
+    }
+
+    return &structs.QueryAggregators{
+        PipeCommandType: structs.GenerateEventType,
+        GenerateEvent: &structs.GenerateEvent{
+            InputLookup: inputLookup,
+        },
+    }, nil
+}
+/ PIPE CMD_INPUTLOOKUP filename:(String) whereClause:(OptionalWhereClause)? {
+    inputLookup:= &structs.InputLookup{}
+    inputLookup.Filename = removeQuotes(filename.(string))
+    if whereClause != nil {
+        inputLookup.WhereExpr = whereClause.(*structs.BoolExpr)
+    }
+    if inputLookup.Max == 0 {
+        inputLookup.Max = 1000000000
+    }
+
+    return &structs.QueryAggregators{
+        PipeCommandType: structs.GenerateEventType,
+        GenerateEvent: &structs.GenerateEvent{
+            InputLookup: inputLookup,
+        },
+    }, nil
+}
+
+InputLookupAggBlock <- inputLookupBlock:InputLookupBlock {
+    inputLookup := inputLookupBlock.(*structs.QueryAggregators)
+    if !inputLookup.GenerateEvent.InputLookup.Append {
+        return nil, fmt.Errorf("Inputlookup after search must have append as true")
+    }
+    inputLookup.GenerateEvent.InputLookup.HasPrevResults = true
+    
+    return inputLookup, nil
+}
```

Comment:
Technically, it could be after any generating command like gentimes, inputlookup, etc. and not specifically after a search so named it as an agg block.

Reply:
Ok, then you should probably update your error message to clarify

Reply:
addressed in the latest commit.